src/app/admin/manageUser/ManageUserModal.tsx

```diff
@@ -0,0 +1,92 @@
+import React, { useState } from "react";
+import { UserRow } from "@/app/admin/page";
+import Modal from "@/components/Modal/Modal";
+import styled from "styled-components";
+import ResetPasswordForm from "@/app/admin/manageUser/ResetPasswordForm";
+import EditUserForm from "@/app/admin/manageUser/EditUserForm";
+import ManageUserOptions from "@/app/admin/manageUser/ManageUserOptions";
+import { AlertOptions, SetAlertOptions } from "@/app/admin/common/SuccessFailureAlert";
+
+const ManageModalContent = styled.div`
+    padding: 0 0.5rem;
+`;
+
+export const EditOption = styled.div`
+    margin-bottom: 1.5rem;
+`;
+
+export const EditHeader = styled.h2`
+    margin-bottom: 0.5rem;
+`;
+
+export type ManageMode = "editDetails" | "resetPassword" | "options";
+
+interface Props {
+    userToEdit: UserRow | null;
+    setUserToEdit: (user: UserRow | null) => void;
```

Comment:
So, the modal is only open when `userToEdit` is not `null`. The `userToEdit` variable is managed in the parent component `UsersTable` which sets the user that should be edited on an edit button click within a particular row.

The `setUserToEdit` function needs to be passed down so that the `ManageUserModal` component can set `userToEdit` to `null` within the parent component once the edit operation has finished and the modal can be closed accordingly with the correct state set in the parent